pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -126,6 +182,11 @@ func newCmdConfigSetAuthInfo(out io.Writer, options *createAuthInfoOptions) *cob
 	cmd.Flags().Var(&options.password, clientcmd.FlagPassword, clientcmd.FlagPassword+" for the user entry in kubeconfig")
 	cmd.Flags().Var(&options.authProvider, flagAuthProvider, "Auth provider for the user entry in kubeconfig")
 	cmd.Flags().StringSlice(flagAuthProviderArg, nil, "'key=value' arguments for the auth provider")
+	cmd.Flags().Var(&options.execCommand, flagExecCommand, "Command for the exec credential plugin for the user entry in kubeconfig")
+	cmd.Flags().Var(&options.execAPIVersion, flagExecAPIVersion, "API version of the exec credential plugin for the user entry in kubeconfig. Defaults to client.authentication.k8s.io/v1beta1")
+	cmd.Flags().StringSlice(flagExecArg, nil, "new arguments for the exec credential plugin command for the user entry in kubeconfig")
```

Comment:
this add/remove style is strange, especially because removal is ambiguous with respect to order. I'd expect the args and env specified to stomp any existing value

Reply:
using StringSlice here means values containing commas cannot be expressed. Should this be StringArray?

Reply:
Yes, StringArray is better here.

I took the add/removal pattern from --auth-provider-arg. It will be a bit illogical if we change it for "exec".
In other words, now --auth-provider-arg adds a new argument without touching the existing values, but you suggest --exec-arg to remove all existing data and add just one value. 
If you think it's okay, then I'll rewrite the code accordingly.
As an option, we can do this only for --exec-arg, and keep the behavior of --auth-provider-arg for --exec-env.

Reply:
> I took the add/removal pattern from --auth-provider-arg

that does not register two flags, it registered a single flag and parses add/remove here:

https://github.com/kubernetes/kubernetes/blob/6671d2556f1af67e703c329b1186896d7c6f9f4d/pkg/kubectl/cmd/config/create_authinfo.go#L255

Following that pattern for envvars is fine. I think specifying the command line entirely and replacing makes more sense.

Reply:
Done